src/components/subscriptions/SubscriptionSubsidyRequests.jsx

```diff
@@ -7,35 +7,59 @@ import SubsidyRequestManagementTable, {
   useSubsidyRequests,
   SUPPORTED_SUBSIDY_TYPES,
   PAGE_SIZE,
+  SUBSIDY_REQUEST_STATUS,
 } from '../SubsidyRequestManagementTable';
+import { ApproveLicenseRequestModal, DeclineSubsidyRequestModal } from '../subsidy-request-modals';
+import EnterpriseAccessApiService from '../../data/services/EnterpriseAccessApiService';
+import { NoAvailableLicensesBanner } from '../subsidy-request-management-alerts';
+import { SubscriptionContext } from './SubscriptionData';
+import LoadingMessage from '../LoadingMessage';
 
 const SubscriptionSubsidyRequests = ({ enterpriseId }) => {
   const {
     isLoading,
     requests,
     requestsOverview,
     handleFetchRequests,
+    updateRequestStatus,
   } = useSubsidyRequests(enterpriseId, SUPPORTED_SUBSIDY_TYPES.licenses);
+  const { data: subscriptionsData, loading: isLoadingSubscriptions } = useContext(SubscriptionContext);
 
-  /* eslint-disable no-console */
-  const handleApprove = (row) => console.log('approve', row);
-  const handleDecline = (row) => console.log('decline', row);
-  /* eslint-enable no-console */
+  const [selectedRequest, setSelectedRequest] = useState();
+  const [isApproveModalOpen, setIsApproveModalOpen] = useState(false);
+  const [isDenyModalOpen, setIsDenyModalOpen] = useState(false);
+
+  if (isLoadingSubscriptions) {
+    return <LoadingMessage className="subscriptions" />;
+  }
+
+  const subscriptions = subscriptionsData.results;
+
+  const hasAvailableLicenses = subscriptions.some(
```

Comment:
nit: similar to other comment, should this have `useMemo` to avoid iteration over the `subscriptions` array on each re-render?